### ISSUE TITLE
KFImage (7.1.0) freezes UI updates on iOS 14, okay on iOS 15

### ISSUE THREAD

**Issue by JetForMe** [onevcat/Kingfisher#1849]
### Check List

Thanks for considering to open an issue. Before you submit your issue, please confirm these boxes are checked.

- [x] I have read the [wiki page](https://github.com/onevcat/Kingfisher/wiki) and [cheat sheet](https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet), but there is no information I need.
- [x] I have searched in [existing issues](https://github.com/onevcat/Kingfisher/issues?utf8=✓&q=is%3Aissue), but did not find a same one.
- [x] I want to report a problem instead of asking a question. It'd better to use [kingfisher tag in Stack Overflow](http://stackoverflow.com/questions/tagged/kingfisher) to ask a question.

### Issue Description

Our app is a mix of UIKit and SwiftUI. We use Kingfisher in both, but SwiftUI makes extensive use of KFImage. We noticed a behavior with our most recent build wherein a section of the app that has a dozen or so KFImages loading, the UI would just freeze and no longer update. Touch events in the UI went ignored, but other operations continued.

Attempting to show the view hierarchy in Xcode 13 resulted in an error message (after a long delay) saying it couldn't get the info needed. Randomly stopping and starting execution in the debugger resulted in many times stopping in KF MD5 hashing code. Examining one of the URLs being loaded let me to a section of my code where I commented out our use of KFImage. Doing so restored proper behavior of the UI.

Reverting to 7.0.0 seems to have fixed the issue. So, the combination of iOS 14.x and 7.1.0 causes the problem. The KFImage invocation looks like this:

```swift
KFImage(self.profileImageURL)
	.cancelOnDisappear(true)
	.placeholder {
		ProfileImagePlaceholderFactory.image(for: self.message.user.username)
			.resizable()
			.aspectRatio(contentMode: .fit)
			.frame(width: kProfileImageSize.width, height: kProfileImageSize.height)
			.foregroundColor(.white)
			.background(Color.purple)
			.clipShape(Circle())
	}
	.resizable()
	.aspectRatio(contentMode: .fill)
	.frame(width: kProfileImageSize.width, height: kProfileImageSize.height)
	.clipShape(Circle())
```

**Comment by benjamincombes** [comment]
Hi, I think I have this issue too, someone on my team reported similar behavior. I am trying to get more information, but he is on iOS 14 too, and I had no problem on my iOS 15 device.

**Comment by atikomtancharoen** [comment]
I found same the issue in iOS 14.5 and Kingfisher 7.1.1
<img width="1024" alt="Screen Shot 2564-11-09 at 22 11 43" src="https://user-images.githubusercontent.com/4318305/140950483-d655c743-2584-416f-b3bc-0cd74edf3d86.png">

<img width="1025" alt="Screen Shot 2564-11-09 at 22 17 52" src="https://user-images.githubusercontent.com/4318305/140951587-6f674335-2894-4fc9-a22b-7ad03f3852f8.png">

**Comment by tatsuz0u** [comment]
@onevcat Probably the same issue found in iOS 15.0 / 15.1 (simulator & physical device) and Kingfisher 7.1.1.
Please try loading multiple images from the Internet, you should see some blanks between images.

```
import SwiftUI
import Kingfisher

struct ContentView: View {
    init() {
        KingfisherManager.shared.cache.clearCache(completion: nil)
    }

    var body: some View {
        ScrollView {
            LazyVGrid(columns: [GridItem(.adaptive(minimum: 100, maximum: 120), spacing: 10)]) {
                ForEach(1..<352) { index in
                    VStack {
                        KFImage(URL(string: "https://github.com/tatsuz0u/Imageset/blob/main/JPGs/\(index).jpg?raw=true"))
                            .placeholder {
                                ZStack {
                                    Color(.systemGray5)
                                    ProgressView()
                                }
                                .aspectRatio(8/11, contentMode: .fill)
                            }
                            .resizable().scaledToFit()
                        Text("\(index)").font(.caption).foregroundColor(.secondary)
                    }
                }
            }
        }
    }
}
```

<img src="https://user-images.githubusercontent.com/31207151/142798728-f28e8a39-8922-4fd5-90e1-3e356b030b15.png" width="200">

**Comment by onevcat** [comment]
@tatsuz0u Thanks for reporting this. Although it should be another unrelated issue, it is fixed in #1873.

**Comment by tatsuz0u** [comment]
> @tatsuz0u Thanks for reporting this. Although it should be another unrelated issue, it is fixed in #1873.

Thanks! I can finally use `.fade()` now!
